### PR TITLE
Ic pomodoro team controller unit tests

### PR DESCRIPTION
**Pull request by ivanCosic**

### CODE REVIEW

File: web/src/test/java/eu/execom/labs/pomodorospringreact/TeamControllerTest.java [hunk 1]
```diff
@@ -0,0 +1,264 @@
...
+        mockMvc = MockMvcBuilders.webAppContextSetup(wac)
+                .addFilter(new CorsFilter())
+                .build();
+    }
+
+    /**
+     * See the method name
+     */
+    @Test
+    public void shouldGetAllTeams() throws Exception {
```
Comment (idostanic) [inline web/src/test/java/eu/execom/labs/pomodorospringreact/TeamControllerTest.java]: why does test method throw Exception?

File: web/src/test/java/eu/execom/labs/pomodorospringreact/TeamControllerTest.java [hunk 2]
```diff
@@ -0,0 +1,264 @@
...
+        userDtoList.add(new UserDto(
+                new User(2, "test1@test.test", "Tester Testerson", null, new HashSet<>(), new HashSet<>())));
+        userDtoList.add(new UserDto(
+                new User(3, "test2@test.test", "Mocker Mockingson", null, new HashSet<>(), new HashSet<>())));
+        userDtoList.add(new UserDto(
+                new User(4, "test3@test.test", "Stuber Stubingson", null, new HashSet<>(), new HashSet<>())));
+
+        return userDtoList;
+    }
+}
```
Comment (idostanic) [inline web/src/test/java/eu/execom/labs/pomodorospringreact/TeamControllerTest.java]: I would remove "should" from every test name.

File: web/src/test/java/eu/execom/labs/pomodorospringreact/TeamControllerTest.java [hunk 3]
```diff
@@ -0,0 +1,264 @@
...
+     * Should return HttpStatus.BAD_REQUEST if name passed is not unique
+     */
+    @Test
+    public void shouldFailToCreateTeamIfNameNotUnique() throws Exception {
+        //given
+        String teamName = "NotUniqueTeamName";
+        given(teamService.createTeam(teamName)).willThrow(new TeamNameNotUniqueException(teamName));
+
+        //when-then
+        mockMvc.perform(post(TEAM_URL_MAPPING + "/new/{name}", "NotUniqueTeamName"))
```
Comment (idostanic) [inline web/src/test/java/eu/execom/labs/pomodorospringreact/TeamControllerTest.java]: why did you pass "NotUniqueTeamName" as new String, when you already have it assigned already?

File: web/src/test/java/eu/execom/labs/pomodorospringreact/TeamControllerTest.java (hunk 1 above)
Comment (ivanCosic) [reply web/src/test/java/eu/execom/labs/pomodorospringreact/TeamControllerTest.java]: MockMvc#perform() throws an exception, and there is no sense to catch it in test methods, so I propagate it further. I've seen in being done this way in many examples, when I investigated how to test REST endpoints with MockMvc.

File: web/src/test/java/eu/execom/labs/pomodorospringreact/TeamControllerTest.java (hunk 3 above)
Comment (ivanCosic) [reply web/src/test/java/eu/execom/labs/pomodorospringreact/TeamControllerTest.java]: My bad, I will fix it.

File: web/src/test/java/eu/execom/labs/pomodorospringreact/TeamControllerTest.java (hunk 1 above)
Comment (idostanic) [reply web/src/test/java/eu/execom/labs/pomodorospringreact/TeamControllerTest.java]: ok

File: web/src/test/java/eu/execom/labs/pomodorospringreact/TeamControllerTest.java (hunk 2 above)
Comment (ivanCosic) [reply web/src/test/java/eu/execom/labs/pomodorospringreact/TeamControllerTest.java]: I've discussed test method naming with @stanislavgrujic when I was writing unit tests for service layer, and he proposed to have self-explanatory method names which should hint what is test expectation, something similar to pattern of shouldDoThis or shouldDoThat...

Comment (idostanic) [reply web/src/test/java/eu/execom/labs/pomodorospringreact/TeamControllerTest.java]: its what I would do, i see its all around code so leave it as it is.

**Pull request merged**